File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/Order/Inventory.cpp
#include "Inventory.h"
#include "Inventory.h"
#include "ProductStock.h"
#include <vector>
#include <string>

namespace Order
{
  Inventory::Inventory()
  {
  }
  void Inventory::addProductItems(int productId, int amount)
  {
    for (auto& productStock : this->products)
    {
      if (productStock->id == productId)
      {
        productStock->amount = productStock->amount + amount;
        break;
      }
    }
  }
  void Inventory::removeProductItems(int productId, int amount)
  {
    for (auto& productStock : this->products)
    {
      if (productStock->id == productId)
      {
        productStock->amount = productStock->amount - amount;
        break;
      }
    }
  }
  std::vector<int> Inventory::getProductIds()
  {
    std::vector<int> result = { };
    for (auto& productStock : this->products)
    {
      result.push_back(productStock->id);
    }
    return result;
  }
  int Inventory::getProductItemsAmount(int productId)
  {
    for (auto& productStock : this->products)
    {
      if (productStock->id == productId)
      {
        return productStock->amount;
      }
    }
    return 0;
  }
  std::string Inventory::getProductName(int productId)
  {
    for (auto& productStock : this->products)
    {
      if (productStock->id == productId)
      {
        return productStock->name;
      }
    }
    return std::string("");
  }
  int Inventory::searchProduct(std::string productName)
  {
    for (auto& productStock : this->products)
    {
      if (productStock->name == productName)
      {
        return productStock->id;
      }
    }
    return 0;
  }
}

File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/Order/Inventory.cs
/// <filename>
///     Inventory.cs
/// </filename>
using System.Collections.Generic;
using Order;

namespace Order
{
    public class Inventory : Order.IInventoryInterface, Order.ISearchInterface
    {

        public Inventory()
        {
        }

        public virtual void addProductItems(int productId, int amount)
        {
            foreach (var productStock in this.products)
            {
                if (productStock.id == productId)
                {
                    productStock.amount = productStock.amount + amount;
                    break;
                }
            }
        }

        public virtual void removeProductItems(int productId, int amount)
        {
            foreach (var productStock in this.products)
            {
                if (productStock.id == productId)
                {
                    productStock.amount = productStock.amount - amount;
                    break;
                }
            }
        }

        public virtual System.Collections.Generic.List<int> getProductIds()
        {
            System.Collections.Generic.List<int> result = new System.Collections.Generic.List<int>();
            foreach (var productStock in this.products)
            {
                result.Add(productStock.id);
            }
            return result;
        }

        public virtual int getProductItemsAmount(int productId)
        {
            foreach (var productStock in this.products)
            {
                if (productStock.id == productId)
                {
                    return productStock.amount;
                }
            }
            return 0;
        }

        public virtual string getProductName(int productId)
        {
            foreach (var productStock in this.products)
            {
                if (productStock.id == productId)
                {
                    return productStock.name;
                }
            }
            return "";
        }

        public virtual int searchProduct(string productName)
        {
            foreach (var productStock in this.products)
            {
                if (productStock.name == productName)
                {
                    return productStock.id;
                }
            }
            return 0;
        }

        public System.Collections.Generic.List<ProductStock> products = new System.Collections.Generic.List<ProductStock>();
    }
}
